File: intellij/src/main/kotlin/dev/nylo/plugin/env/EnvFile.kt
package dev.nylo.plugin.env

import java.io.File

data class EnvFile(
    val file: File,
    val suffix: String?,
    val displayName: String,
) {
    val fileName: String = file.name

    val toolActionId: String = "Tool_External Tools_${MetroToolNaming.toolName(displayName)}"
}

object MetroToolNaming {
    fun toolName(displayName: String): String = "Metro make:env $displayName"
}

File: vscode/src/core/env/envFile.ts
/** Port of `dev.nylo.plugin.env.EnvFile` (the data shape; IDE-specific ids are dropped). */
export interface EnvFile {
  /** Absolute path to the env file. */
  filePath: string;
  /** Bare file name, e.g. `.env.prod`. */
  fileName: string;
  /** Suffix after `.env.`, e.g. `prod.staging`; `null` for a bare `.env`. */
  suffix: string | null;
  /** Title-cased display name, e.g. `Prod Staging` / `Default`. */
  displayName: string;
}
